### PR TITLE
more e2e tests

### PR DESCRIPTION
**Pull request by ortex**
add e2e TestEnd2End_Blocking for ClockInternal clock source; split default e2e test for with/without fec

#102 #104

### CODE REVIEW

File: roc/e2e_test.go [hunk 1]
```diff
@@ -86,78 +103,181 @@ func (e *e2e) close(t *testing.T) {
...
-					t.Fail()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := newE2E(t, tt.params)
+			defer e.close(t)
+
+			samplesCnt := 100
+			testSamples := generateTestSamples(samplesCnt)
+
+			var interval = time.Second / time.Duration(44100/samplesCnt)
```
Comment (gavv) [inline roc/e2e_test.go:135]: nit: Let's use sample rate from config, instead of 44100?

**Review by gavv — COMMENTED** [review]
Thanks! One small comment, not even related to PR itself.

File: roc/e2e_test.go (hunk 1 above)
Comment (ortex) [reply roc/e2e_test.go]: done!

**Comment by ortex** [comment]
@gavv Could you have a look?

fixed the discussion and set go-version to 1.20 instead of the latest because our ci pipeline fails with new go 1.21, see https://github.com/roc-streaming/roc-go/actions/runs/6067814352 I will create a separate issue for that

Btw, need to use '1.20' as go version for action instead of 1.20, because 1.20 is treated as 1.2 https://github.com/actions/setup-go/issues/326#issuecomment-1561222838

**Comment by gavv** [comment]
Thanks! Damn, it was float!

**Pull request merged**